module/VuFind/src/VuFind/Hierarchy/Driver/PluginManager.php

```diff
@@ -38,6 +38,27 @@
  */
 class PluginManager extends \VuFind\ServiceManager\AbstractPluginManager
 {
+    /**
+     * Default plugin aliases.
+     *
+     * @var array
+     */
+    protected $aliases = [
+        'default' => 'VuFind\Hierarchy\Driver\HierarchyDefault',
+        'flat' => 'VuFind\Hierarchy\Driver\HierarchyFlat',
+    ];
+
+    /**
+     * Default plugin factories.
+     *
+     * @var array
+     */
+    protected $factories = [
+        'VuFind\Hierarchy\Driver\HierarchyDefault' =>
+            'VuFind\Hierarchy\Driver\Factory',
```

Comment:
I would call this ConfigurationBasedFactory, rather than just Factory, to tie it to the class it constructs. That way, if we create other classes and other factories in future, it will be less confusing.